fix(storybook): voeg ambient module declaration toe voor *.mdx imports

## User Story

Als ontwikkelaar wil ik dat `tsc --noEmit` geen TS2307-fouten geeft op `.mdx` imports, zodat de TypeScript check betrouwbare output geeft en toekomstige echte fouten niet verdrinken in ruis.

## Context

De Storybook package importeert `.mdx` bestanden als React componenten:

\`\`\`ts
import DocsPage from './Button.docs.mdx';
\`\`\`

Vite en Storybook verwerken deze imports correct op build-tijd. Maar `tsc --noEmit` kent `.mdx` niet als module en gooit voor **36 story-bestanden** een TS2307-fout:

```
Cannot find module './Button.docs.mdx' or its corresponding type declarations.
```

Er is al een precedent in het project: `packages/components-react/src/svg.d.ts` doet hetzelfde voor `*.svg?react` imports.

**Fix**: één nieuw bestand `packages/storybook/src/mdx.d.ts` met:

\`\`\`ts
declare module '*.mdx' {
  import type { ComponentType } from 'react';
  const component: ComponentType;
  export default component;
}
\`\`\`

Dit bestand valt automatisch binnen de `"include": ["src/**/*"]` van `packages/storybook/tsconfig.json` — geen config-wijzigingen nodig.

**Scope**: alleen `packages/storybook/src/mdx.d.ts` — één nieuw bestand, nul bestaande bestanden aangepast.

## Acceptance Criteria

- [ ] `packages/storybook/src/mdx.d.ts` bestaat met ambient `*.mdx` module declaration
- [ ] `pnpm --filter storybook exec tsc --noEmit` geeft geen TS2307-fouten meer op `.mdx` imports (was 36 fouten)
- [ ] Geen bestaande bestanden zijn aangepast

## Definition of Done

### Voorbereiding

- [ ] Feature branch aangemaakt: `git checkout -b fix/mdx-type-declaration`

### Werkzaamheden

- [ ] Implementatie afgerond

### Kwaliteitscontrole

- [ ] Tests groen: `pnpm test`
- [ ] TypeScript schoon: `pnpm --filter storybook exec tsc --noEmit`
- [ ] Lint schoon: `pnpm lint`

### Oplevering

- [ ] PR aangemaakt: `gh pr create`
- [ ] CI groen op de branch
- [ ] PR gemerged: `gh pr merge --merge`

## Notities / Open vragen

- Bewust geen `{ Meta, Story, ... }` named exports gedeclareerd in de module declaration — die komen van `@storybook/blocks`, niet van de `.mdx` bestanden zelf.
- Het `?raw` query-patroon (voor `.docs.md` imports) werkt al correct en heeft geen eigen declaration nodig.

--- a/packages/storybook/src/mdx.d.ts
+++ b/packages/storybook/src/mdx.d.ts
@@ -1,0 +1,5 @@
+declare module '*.mdx' {
+  import type { ComponentType } from 'react';
+  const component: ComponentType;
+  export default component;
+}

